### PR TITLE
Add new cit_ref_label style for generating citation_reference nodes

### PR DESCRIPTION
**Pull request by mcmtroffaes**
See issue #275 

@brechtm Could you give this branch a spin? You'll need to render with the cit_ref_label style. As a test document, to check that all citations are correctly rendered, you can use https://github.com/mcmtroffaes/sphinxcontrib-bibtex/tree/feature/generate-citation-reference-2/test/roots/test-citation_roles_2

### CODE REVIEW

**Comment by brechtm** [comment]
This works indeed when `bibtex_reference_style = 'cit_ref_label'`. But I don't see why only this style would be supported when using the rinoh builder.

File: src/sphinxcontrib/bibtex/style/template.py [hunk 1]
```diff
@@ -162,6 +162,75 @@ def reference(children, data: Dict[str, Any]):
...
+        targetid: str,
+        child: Union[Node, List[Node]], title: str = None
+        ) -> docutils.nodes.citation_reference:
+    """Shortcut to create a citation reference node."""
+    # latex builder needs docname and refname
+    node = docutils.nodes.citation_reference('', '', internal=True)
+    if builder.name in {'latex', 'rinoh'}:
+        # builders using docname/refname
+        node['docname'] = todocname
+        node['refname'] = targetid
```
Comment (mcmtroffaes) [inline src/sphinxcontrib/bibtex/style/template.py:178]: @brechtm You could check here if rinoh builder needs special casing in the attributes it needs for the citation_reference to resolve correctly. I've assumed it works like the latex builder.

**Comment by mcmtroffaes** [comment]
> This works indeed when `bibtex_reference_style = 'cit_ref_label'`. 

Great! Glad this worked.

> But I don't see why only this style would be supported when using the rinoh builder.

If you use citation_reference nodes, sphinx/docutils forcibly adds brackets around them. That obviously breaks the rest of the style (which normally takes care of bracket placement itself, not necessarily directly around the reference, e.g. multiple references can be placed by the style inside the same bracket with commas in between, and more). The only way I can see this solved is having a separate style which knows not to place brackets around citation_reference nodes.

If the rinoh builder can deal with citation references from regular reference nodes (like the html builder), then rinoh should be able to deal with all citation styles.

**Comment by mcmtroffaes** [comment]
> But at that point it's much simpler to just add the 6 lines of code from https://github.com/mcmtroffaes/sphinxcontrib-bibtex/issues/275#issuecomment-1079075883, no? All that would be needed then, is somehow instructing rinohtype to not add the brackets.

Sure, that's much simpler. The refid looks non-standard (starting with a percent sign); is that rinoh specific?

**Comment by mcmtroffaes** [comment]
Superseded by #291

**Pull request closed without merging**